### PR TITLE
initial code for registering operator to olm

### PR DESCRIPTION
**Pull request by animeshsingh**
starter metadata to allow operator to be registered to operatorhub and to olm, and be orchestrated from there

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kfctl/208)

### CODE REVIEW

File: deploy/olm-catalog/kfctl/0.1.0/kfctl.v0.1.0.clusterserviceversion.yaml [hunk 1]
```diff
@@ -0,0 +1,86 @@
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+metadata:
+  annotations:
+    alm-examples: '[]'
+    capabilities: Basic Install
+  name: kfctl.v0.1.0
+  categories: "Machine Learning"
```
[the remark this answers is not in the document]
Comment (animeshsingh) [reply deploy/olm-catalog/kfctl/0.1.0/kfctl.v0.1.0.clusterserviceversion.yaml]: addressed @Tomcli

File: deploy/olm-catalog/kfctl/0.1.0/kfctl.v0.1.0.clusterserviceversion.yaml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    alm-examples: '[]'
+    capabilities: Basic Install
+    categories: "Machine Learning"
+    description: "Kubeflow Operator for deployment and management of Kubeflow"
+    support: Kubeflow
+  name: kfctl.v0.1.0
+  namespace: placeholder
+spec:
+  apiservicedefinitions: {}
+  customresourcedefinitions: {}
```
Comment (Tomcli) [inline deploy/olm-catalog/kfctl/0.1.0/kfctl.v0.1.0.clusterserviceversion.yaml]: `owned` field is missing from `customresourcedefinitions`. You can refer to the [operator example](https://github.com/operator-framework/operator-sdk-samples/blob/master/go/memcached-operator/deploy/olm-catalog/memcached-operator/0.12.0/memcached-operator.v0.12.0.clusterserviceversion.yaml#L24-L59) to fill in the necessary fields. 

It looks like some of these generated code is missing during `operator-sdk generate csv`.  The [command](https://github.com/operator-framework/operator-sdk/blob/master/internal/scaffold/olm-catalog/csv_updaters.go#L245-L248) is looking for the go apis spec under the defined [ApisDir](https://github.com/operator-framework/operator-sdk/blob/master/internal/scaffold/constants.go#L29) which is a constant looking the apis under /pkg/apis/kfdef where the actual kfdef's apis in kfctl is /pkg/apis/app/kfdef. 

Probably we should open an issue on Operator-SDK to suggest some of these variables to be configurable for operators that are generated on an existing package.

File: deploy/olm-catalog/kfctl/0.1.0/kfctl.v0.1.0.clusterserviceversion.yaml [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+      url: https://github.com/kubeflow
+  maintainers:
+    - name: Animesh Singh
+      email: singhan@us.ibm.com
+    - name: Tommy Li
+      email: tommy.chaoping.li@ibm.com
+    - name: Weiqiang Zhuang
+      email: wzhuang@us.ibm.com
+  replaces: kfctl.v0.0.0
+  version: 0.1.0
```
Comment (Tomcli) [inline deploy/olm-catalog/kfctl/0.1.0/kfctl.v0.1.0.clusterserviceversion.yaml]: missing `keywords` field under [spec](https://github.com/operator-framework/operator-sdk-samples/blob/master/go/memcached-operator/deploy/olm-catalog/memcached-operator/0.12.0/memcached-operator.v0.12.0.clusterserviceversion.yaml#L165-L167).

**Comment by Tomcli** [comment]
Related issue: https://github.com/operator-framework/operator-sdk/issues/2493

File: deploy/olm-catalog/kfctl/0.1.0/kfctl.v0.1.0.clusterserviceversion.yaml (hunk 2 above)
Comment (animeshsingh) [reply deploy/olm-catalog/kfctl/0.1.0/kfctl.v0.1.0.clusterserviceversion.yaml]: Thanks Tommy for the pointer to the OLM bug, and potential fix.- this should be updated now

**Comment by vpavlin** [comment]
@LaVLaS can you check this PR as our Open Data Hub CSV expert:)

**Comment by LaVLaS** [comment]
I ran a quick verification based on OperatorHub [requirements](https://github.com/operator-framework/community-operators/blob/master/docs/required-fields.md) and [Testing Operator Deployment on OpenShift](https://github.com/operator-framework/community-operators/blob/master/docs/testing-operators.md#testing-operator-deployment-on-openshift). I had to make a few changes to get a quay.io application successfully deploying via the OpenShift OperatorHub

- The CRD is missing from the kfctl 0.1.0 version folder.
- CSV File
  - Changed `metadata.annotations.categories` to `AI/Machine Learning` so that it adheres to OperatorHub [categories list](https://github.com/operator-framework/community-operators/blob/master/docs/required-fields.md#categories)
  -  Changed `spec.provider` to be a dictionary with single key `name: Kubeflow`
    The `url` value in this field could be moved to `metadata.annotations.repository`

These were just some quick changes to test it as a quay application that could be automatically pulled in the from a quay.io application

File: deploy/role.yaml [hunk 4]
```diff
@@ -0,0 +1,20 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  annotations:
+    rbac.authorization.kubernetes.io/autoupdate: "true"
+  creationTimestamp: null
+  labels:
+    kubernetes.io/bootstrapping: rbac-defaults
+  name: cluster-admin
```
Comment (Tomcli) [inline deploy/role.yaml]: This needs to be `kubeflow-operator` in order to match with the operator

File: deploy/olm-catalog/kfctl/0.1.0/kfctl.v0.1.0.clusterserviceversion.yaml [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+          - '*'
+          resources:
+          - '*'
+          verbs:
+          - '*'
+        - nonResourceURLs:
+          - '*'
+          verbs:
+          - '*'
+        serviceAccountName: cluster-admin
```
Comment (Tomcli) [inline deploy/olm-catalog/kfctl/0.1.0/kfctl.v0.1.0.clusterserviceversion.yaml]: This needs to be `kubeflow-operator` after the code gen.

File: deploy/role.yaml (hunk 4 above)
Comment (animeshsingh) [reply deploy/role.yaml]: done

File: deploy/olm-catalog/kfctl/0.1.0/kfctl.v0.1.0.clusterserviceversion.yaml (hunk 5 above)
Comment (animeshsingh) [reply deploy/olm-catalog/kfctl/0.1.0/kfctl.v0.1.0.clusterserviceversion.yaml]: done

**Comment by animeshsingh** [comment]
> 
> * The CRD is missing from the kfctl 0.1.0 version folder.
> * CSV File
>   
>   * Changed `metadata.annotations.categories` to `AI/Machine Learning` so that it adheres to OperatorHub [categories list](https://github.com/operator-framework/community-operators/blob/master/docs/required-fields.md#categories)
>   * Changed `spec.provider` to be a dictionary with single key `name: Kubeflow`
>     The `url` value in this field could be moved to `metadata.annotations.repository`
> 

Addressed these @LaVLaS

**Comment by animeshsingh** [comment]
> I ran a quick verification based on OperatorHub [requirements](https://github.com/operator-framework/community-operators/blob/master/docs/required-fields.md) and [Testing Operator Deployment on OpenShift](https://github.com/operator-framework/community-operators/blob/master/docs/testing-operators.md#testing-operator-deployment-on-openshift). I had to make a few changes to get a quay.io application successfully deploying via the OpenShift OperatorHub

Added the link to those instructions in the main Operator.md file @LaVLaS

**Comment by animeshsingh** [comment]
@gabrielwen @jlewi this is good to go

**Pull request merged**